packages/flutter/lib/src/rendering/object.dart

```diff
@@ -987,8 +1008,20 @@ class PipelineOwner {
     try {
       final List<RenderObject> dirtyNodes = _nodesNeedingPaint;
       _nodesNeedingPaint = <RenderObject>[];
+      // Need to reset the layer of every repaint boundary; otherwise, a child
+      // layer may be appended to a new parent layer before it is removed from
+      // the old parent layer.
+      dirtyNodes.sort((RenderObject a, RenderObject b) => b.depth - a.depth);
+      for (final RenderObject node in dirtyNodes) {
```

Comment:
Since this is performance critical code: We are not doing any extra work now, right? We are just doing the reset upfront instead of later?

Reply:
yes, but it becomes two for-loops instead of one. Not sure whether there will be any impact on perf. If this is not acceptable, I will then have to do something similar to globalkey duplication that temporary allows child to be attached to two parents and adds a postframecallback to make sure the previous parent give up the child.

This will be more complicated than the global key cases though, since it involves detaching/attaching logic.